### PR TITLE
[Tests-Only] Added test scenario for searching file/folders by tags

### PR DESCRIPTION
**Pull request by kiranparajuli589**
## Description
This PR adds tests for search using tags from WebDAV API

## Related Issue

- Part of https://github.com/owncloud/QA/issues/634

## Motivation and Context
More `api-test` coverage with **REPORT** method

## How Has This Been Tested?

- :robot: 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

File: tests/acceptance/features/bootstrap/SearchContext.php [hunk 1]
```diff
@@ -38,6 +38,42 @@ class SearchContext implements Context {
...
+			= "<?xml version='1.0' encoding='utf-8' ?>\n" .
+			"	<oc:filter-files xmlns:d='DAV:' xmlns:oc='http://owncloud.org/ns' >\n" .
+			"		<d:prop>\n" .
+			"			<oc:id/>\n" .
+			"		</d:prop>\n" .
+			"		<oc:filter-rules>\n" .
+			"			<oc:systemtag>$systemTagId</oc:systemtag>\n" .
+			"		</oc:filter-rules>\n" .
+			"	</oc:filter-files>";
+		var_dump($body);
```
Comment (HariBhandari07) [inline tests/acceptance/features/bootstrap/SearchContext.php]: get rid of var dump

**Comment by HariBhandari07** [comment]
Coding Standard is failing

File: tests/acceptance/features/bootstrap/SearchContext.php [hunk 2]
```diff
@@ -38,6 +38,42 @@ class SearchContext implements Context {
...
+	 * @param {string} $user
+	 * @param {string} $tagName
+	 *
+	 * @return void
+	 */
+	public function userSearchesForTagUsingWebDavAPI($user, $tagName) {
+		$baseUrl = $this->featureContext->getBaseUrl();
+		$password = $this->featureContext->getPasswordForUser($user);
+		$response = $this->tagsContext->requestTagsForUser($user, $tagName);
+		var_dump($response->asXML());
```
Comment (HariBhandari07) [inline tests/acceptance/features/bootstrap/SearchContext.php]: get rid of var_dump

File: tests/acceptance/features/bootstrap/SearchContext.php [hunk 3]
```diff
@@ -38,6 +38,42 @@ class SearchContext implements Context {
...
+
+	/**
+	 * @When user :user searches for tag :tagName using the webDAV API
+	 *
+	 * @param {string} $user
+	 * @param {string} $tagName
+	 *
+	 * @return void
+	 */
+	public function userSearchesForTagUsingWebDavAPI($user, $tagName) {
```
Comment (dpakach) [inline tests/acceptance/features/bootstrap/SearchContext.php]: Why is this function not in the `TagsContext.php` ?

Comment (kiranparajuli589) [reply tests/acceptance/features/bootstrap/SearchContext.php]: @dpakach what if I create a new feature inside `apiTags` with name like `searchTags`, we can add more tests for searching by tags using API

Comment (kiranparajuli589) [reply tests/acceptance/features/bootstrap/SearchContext.php]: function moved..

File: tests/acceptance/features/bootstrap/SearchContext.php (hunk 1 above)
Comment (kiranparajuli589) [reply tests/acceptance/features/bootstrap/SearchContext.php]: done!

File: tests/acceptance/features/bootstrap/TagsContext.php [hunk 4]
```diff
@@ -1626,6 +1626,51 @@ public function theAdministratorRemovesTheTagFromFileSharedByUsingTheWebdavApi(
...
+		}
+		$found = \in_array($tagName, $tagNames);
+		if ($found) {
+			$index = \array_search($tagName, $tagNames);
+			$body .=
+				"			<oc:systemtag>$tagIds[$index]</oc:systemtag>\n";
+		} else {
+			throw new Error(
+				"Expected: Tag with name $tagName to be in created list, but not found!" .
+				"List of created Tags: " . \implode(",", $createdTagsArray)
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/TagsContext.php]:
```
    When user "user0" searches for tag "Ooooo" using the webDAV API               # TagsContext::userSearchesForTagUsingWebDavAPI()
      Notice: Array to string conversion in /home/phil/git/owncloud/core/tests/acceptance/features/bootstrap/TagsContext.php line 1658
```
If I test for a tag that does not exist then I get this PHP notice.
IMO `$createdTagsArray` is not an array of strings, it is an array of arrays that (eventually) have strings.

**Review by phil-davis — COMMENTED** [review]
Get this working nicely, then we can merge it.

After that, in another PR, you can add some scenarios that have files in a share that are tagged by the sharer, and the sharee finds them by tag, files in sub-directories and so on.

And then there can be a test step that lets you search for multiple tags (files that have all the requested tags)

File: tests/acceptance/features/bootstrap/TagsContext.php (hunk 4 above)
Comment (kiranparajuli589) [reply tests/acceptance/features/bootstrap/TagsContext.php]: yes, small rabbit hole :smiley: 
Actually I should use `$tagNames` instead `$createdTagsArray`

File: tests/acceptance/features/apiWebdavOperations/search.feature [hunk 5]
```diff
@@ -206,3 +206,15 @@ Feature: Search
...
+    Given user "user0" has created a "normal" tag with name "JustARegularTagName"
+    And user "user0" has added tag "JustARegularTagName" to folder "फनी näme"
+    And user "user0" has added tag "JustARegularTagName" to file "upload.txt"
+    And user "user0" has added tag "JustARegularTagName" to file "a-image.png"
+    When user "user0" searches for tag "JustARegularTagName" using the webDAV API
+    Then the HTTP status code should be "207"
+    And the search result by tags for user "user0" should contain these entries:
+      | फनी näme    |
+      | upload.txt  |
+      | a-image.png |
```
[the remark this answers is not in the document]
Comment (kiranparajuli589) [reply tests/acceptance/features/apiWebdavOperations/search.feature]: done!

**Review by dpakach — COMMENTED** [review]
LGMT :+1:

**Review by phil-davis — APPROVED** [review]

**Comment by individual-it** [comment]
@phil-davis commits should have been squashed before merging

**Pull request merged**